services:
  auth-service-app:
    build:
      context: ./
      dockerfile: ./Dockerfile
    image: auth-service-app
    container_name: auth-service-app
    volumes:
      - .:/var/www/html:cached
      - ./vendor:/var/www/html/vendor:delegated
    environment:
      - APP_ENV=local
      - APP_DEBUG=true
      - DB_CONNECTION=mysql
      - DB_HOST=auth-service-mysql
      - DB_PORT=3306
      - DB_DATABASE=auth_service
      - DB_USERNAME=root
      - DB_PASSWORD=root
    depends_on:
      auth-service-mysql:
        condition: service_healthy
    entrypoint: /usr/local/bin/entrypoint.sh

  auth-service-mysql:
    image: mysql:8.0
    container_name: auth-service-mysql
    ports:
      - "3306:3306"
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_DATABASE: auth_service
    volumes:
      - auth_service_db_data:/var/lib/mysql
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      interval: 10s
      timeout: 5s
      retries: 3

  auth-service-phpmyadmin:
    image: phpmyadmin/phpmyadmin:latest
    container_name: auth-service-phpmyadmin
    ports:
      - "8080:80"
    environment:
      PMA_HOST: auth-service-mysql
      PMA_PORT: 3306
      PMA_USER: root
      PMA_PASSWORD: root
    depends_on:
      auth-service-mysql:
        condition: service_healthy
    platform: linux/amd64

  auth-service-nginx:
    image: nginx:latest
    container_name: auth-service-nginx
    ports:
      - "8000:80"
    volumes:
      - .:/var/www/html
      - ./nginx.conf:/etc/nginx/conf.d/default.conf
    depends_on:
      - auth-service-app

volumes:
  auth_service_db_data:
    driver: local
